src/Device/AndroidSensors.cpp

```diff
@@ -287,6 +288,49 @@ DeviceDescriptor::OnHeartRateSensor(unsigned bpm) noexcept
   e.Commit();
 }
 
+void
+DeviceDescriptor::OnEngineSensors(uint16_t cht_temp,
+                                      uint16_t egt_temp,
+                                      uint16_t oat_temp,
+                                      uint32_t atmospheric_pressure,
+                                      uint16_t revs_per_sec) noexcept
+{
+  const auto e = BeginEdit();
+  NMEAInfo &basic = *e;
+  basic.UpdateClock();
+  basic.alive.Update(basic.clock);
+  basic.engine_state.revs_per_sec = revs_per_sec;
+  basic.engine_state.revs_per_sec_available.Update(basic.clock);
+  // 0 pascal is a invalid value.
```

Comment:
The documentation for the API contract does not belong into an arbitrary implementation of the interface, but in the interface API documentation.